packages/docs/integrate/custom-integration.md

```diff
@@ -0,0 +1,334 @@
+
+export default integrationPlugin(({ integration }) => {
+  integration.configure('jsonplaceholder', { ...<%= serialize(options) %>})
+});
+
+```
+
+Next, add an `index.js` file for your module:
+
+```jsx
+// integrations/jsonplaceholder/nuxt/index.js
+
+import * as path from 'path';
+
+export default function jsonplaceholder(moduleOptions) {
+  this.addPlugin({
+    src: path.resolve(__dirname, 'plugin.js'),
+    options: moduleOptions,
+  });
+}
+```
+
+You're done with the `/nuxt` directory. At this point, your integration is an empty shell and delivers no functionality, but you can register it in your project already!
+
+## Registering custom integration
+
+Every integration has to be registered in both `nuxt.config.js` and `middleware.config.js`.
+
+First, register your newly-created module in `nuxt.config.js`:
+
+```jsx
+// nuxt.config.js
+
+export default {
+  buildModules: ['~/integrations/jsonplaceholder/nuxt'],
+};
+```
+
+::: tip Why `buildModules` and not `modules`?
+
+In this example, you've created a very simple module responsible only for registering your integration's plugin. It does it during the application build, therefore its code does not need to be included in the final bundle.
+
+However, your integration's module could do way more that that. Visit the official Nuxt [documentation](https://nuxtjs.org/docs/directory-structure/modules/) to explore all capabilities of Nuxt modules and the difference between `modules` and `buildModules`.
+:::
+
+Secondly, add your integration's configuration in `middleware.config.js`:
+
+```jsx
+// middleware.config.js
+
+const path = require('path');
+
+module.exports = {
+  integrations: {
+    jsonplaceholder: {
+      location: path.resolve(
+        __dirname,
+        'integrations/jsonplaceholder/src/index.server.js'
+      ),
+      configuration: {
+        baseURL: 'https://jsonplaceholder.typicode.com/',
+        // apiKey: '1234'
+      },
+    },
+  },
+};
+```
+
+In this example, the configuration will only contain `baseURL`. However, if your integration requires passing some sensitive credentials (like tokens or API keys), this is the right place to put them.
+
+You might be wondering about the `location` part of the above configuration, and you are right - it refrences a file that does not exist yet. Do not worry. You're going to create it in the next step.
+
+## Creating an API client
+
+Your integration will need an API client to communicate with the 3rd party. For most services, you would use a client provided by their SDKs. This is what Vue Storefront does for the majority of our integrations (e.g. [Contentstack](https://www.contentstack.com/docs/developers/javascript-browser/api-reference/)).
+
+For [JSON Placeholder Free Fake API](https://jsonplaceholder.typicode.com/), we're simply going to create a dedicated [axios](https://github.com/axios/axios) instance. Let’s start with adding the missing `index.server.js` file in `/integrations/jsonplaceholder/src`.
```

Comment:
```suggestion
For `JSON Placeholder Free Fake API`, we're going to create a dedicated [axios](https://github.com/axios/axios) instance. Let’s start with adding the missing `index.server.js` file in `/integrations/jsonplaceholder/src`.
```

The rule of thumb is to provide a link to service only the first time it's mentioned in the document.